# Week 2 Checkpoint: First Data Profile
**BAN 6003 Data Management and Analytics Integration**

Complete this notebook using the `employee_data.csv` file in the `data/` folder.

This is the independent checkpoint for Week 2. Use the functions practiced in the guided lab. You are not expected to clean the data yet. Your goal is to profile the data and report what you notice.


## Instructions

1. Run each cell from top to bottom.
2. Fill in all cells marked `# Your code here`.
3. Write short answers in the markdown response areas.
4. Save, commit, and push your notebook to GitHub.

You may use the Week 2 guided lab as a reference.


## Task 0: Import Pandas and Load the Data
Run the cell below.


In [1]:
import pandas as pd

df = pd.read_csv("../data/employee_data.csv")
df.head()


,employee_id,first_name,last_name,hire_date,department,salary,age,full_time
0,101,John,Smith,4/15/22,Marketing,85000,34.0,True
1,102,Jane,Doe,8/20/21,Sales,92000,45.0,True
2,103,Peter,Jones,1/2/22,HR,71000,29.0,True
3,104,Mary,Johnson,11/10/23,Engineering,115000,38.0,True
4,105,Mike,Brown,5/30/20,sales,89000,NaN,False


## Task 1: Basic Structure
Show the basic structure of the DataFrame.

Your code should show:
- the shape of the DataFrame;
- the column names;
- the first five rows;
- the last five rows.


In [2]:
# Your code here
print("Shape:", df.shape)
print("\nColumns:", list(df.columns))

display(df.head())
display(df.tail())


Shape: (13, 8)

Columns: ['employee_id', 'first_name', 'last_name', 'hire_date', 'department', 'salary', 'age', 'full_time']


,employee_id,first_name,last_name,hire_date,department,salary,age,full_time
0,101,John,Smith,4/15/22,Marketing,85000,34.0,True
1,102,Jane,Doe,8/20/21,Sales,92000,45.0,True
2,103,Peter,Jones,1/2/22,HR,71000,29.0,True
3,104,Mary,Johnson,11/10/23,Engineering,115000,38.0,True
4,105,Mike,Brown,5/30/20,sales,89000,NaN,False


,employee_id,first_name,last_name,hire_date,department,salary,age,full_time
8,102,Emily,White,1/15/24,Sales,91000,31.0,True
9,110,David,Miller,7/22/19,HR,68000,58.0,True
10,111,Sarah,Moore,3/12/23,Marketing,"$91,000",28.0,True
11,error,Kevin,Clark,2/5/24,Sales,88000,35.0,True
12,113,Brian,Hall,6/18/22,Engineering,5900000,999.0,True


### Task 1 Written Response
Write 1–2 sentences describing the size and general structure of the dataset.

**Your response:**  
The dataset contains 8 columns of employee information and 13 rows that include numeric and categorical variables. It is structed for analyzing work characteristics.


## Task 2: Data Types and Non-Missing Counts
Use Pandas to inspect data types and non-missing counts.

Your code should use:
- `info()`
- `dtypes`


In [3]:
# Your code here# Display data types and non-missing counts
df.info()

# Display data types only
print("\nData Types:")
print(df.dtypes)


<class 'pandas.DataFrame'>
RangeIndex: 13 entries, 0 to 12
Data columns (total 8 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   employee_id  13 non-null     str    
 1   first_name   13 non-null     str    
 2   last_name    13 non-null     str    
 3   hire_date    12 non-null     str    
 4   department   13 non-null     str    
 5   salary       12 non-null     str    
 6   age          12 non-null     float64
 7   full_time    12 non-null     object 
dtypes: float64(1), object(1), str(6)
memory usage: 964.0+ bytes

Data Types:
employee_id        str
first_name         str
last_name          str
hire_date          str
department         str
salary             str
age            float64
full_time       object
dtype: object


### Task 2 Written Response
Which columns, if any, have data types that might need to be investigated later? For example, is a number or date stored as text/object?

**Your response:**  
The hire_date column is stored as str rather than date data type. The salary column is stored as str even though it is numeric. The full_time column is stored as a object type. These should be investigated.


## Task 3: Missing Values
Check missing values for all columns.

Your code should use `isna().sum()`.

Optional: also calculate missing percentages.


In [6]:
# Count missing values in each column
print("Missing Values by Column:")
print(df.isna().sum())

# Calculate missing percentages
print("Missing Percentage by Column:")
print((df.isna().sum() / len(df)) * 100)


Missing Values by Column:
employee_id    0
first_name     0
last_name      0
hire_date      1
department     0
salary         1
age            1
full_time      1
dtype: int64
Missing Percentage by Column:
employee_id    0.000000
first_name     0.000000
last_name      0.000000
hire_date      7.692308
department     0.000000
salary         7.692308
age            7.692308
full_time      7.692308
dtype: float64


### Task 3 Written Response
Which columns have missing values? Which missing values seem most important to investigate later?

**Your response:**  
hire_date, salary, age, and full_time are missing values. salary and age seem the most important to investigate since they are numerical.


## Task 4: Numerical Summary
Use `.describe()` to summarize numerical columns.

Then choose one numerical column and calculate at least two individual summary statistics such as mean, median, min, max, or standard deviation.


In [7]:
# Summary statistics for numerical columns
print(df.describe())

# Choose one numerical column (age) and calculate additional statistics
print("\nAge Summary Statistics:")
print("Mean:", df["age"].mean())
print("Median:", df["age"].median())
print("Minimum:", df["age"].min())
print("Maximum:", df["age"].max())
print("Standard Deviation:", df["age"].std())


              age
count   12.000000
mean   118.583333
std    277.411889
min     28.000000
25%     32.500000
50%     36.500000
75%     46.750000
max    999.000000

Age Summary Statistics:
Mean: 118.58333333333333
Median: 36.5
Minimum: 28.0
Maximum: 999.0
Standard Deviation: 277.41188882347143


### Task 4 Written Response
What do you notice from the numerical summaries? Mention at least one range, unusual value, or pattern.

**Your response:**  
The range of the numerical summaries is quite large with from 28 to 999. 999 is an unusual value and is a possible error since that is not a living age. A pattern I can see from this is the 999 value increases the mean age greatly above the median age value, so this should be checked before continuing.


## Task 5: Categorical Summary
Use `.value_counts()` to summarize at least one categorical column.

Start with `department` if it exists in the dataset. Then choose one additional categorical column if available.


In [ ]:
# Department frequency counts
print("Department Counts:")
print(df["department"].value_counts())

# Full-time status frequency counts
print("Full-Time Status Counts:")
print(df["full_time"].value_counts())


Department Counts:
department
Sales          4
Marketing      3
Engineering    3
HR             2
sales          1
Name: count, dtype: int64

Full-Time Status Counts:
full_time
True     11
False     1
Name: count, dtype: int64


### Task 5 Written Response
What are the most common categories? Do you notice anything that might need standardization or further investigation later?

**Your response:**  
The most common categories are full_time and department. Sales and sales appear under the department and should be investigated as it needs standardization.


## Task 6: Duplicate Checks
Check for duplicate records and duplicate employee IDs.

Your code should check:
- fully duplicated rows;
- duplicate `employee_id` values;
- the most common `employee_id` values using `value_counts()`.


In [9]:
# Count fully duplicated rows
print("Number of Duplicate Rows:")
print(df.duplicated().sum())

# Check for duplicate employee IDs
print("Number of Duplicate Employee IDs:")
print(df["employee_id"].duplicated().sum())

# Show employee ID frequencies
print("Employee ID Counts:")
print(df["employee_id"].value_counts())

Number of Duplicate Rows:
0
Number of Duplicate Employee IDs:
1
Employee ID Counts:
employee_id
102      2
101      1
103      1
104      1
105      1
106      1
107      1
108      1
110      1
111      1
error    1
113      1
Name: count, dtype: int64


### Task 6 Written Response
Are there duplicate employee IDs? If yes, what might that mean if `employee_id` is supposed to uniquely identify each employee?

**Your response:**  
Yes, 102 is duplicated. This means the duplicates indicate duplicate records.


## Task 7: Overall Data Quality Observations
Write 3–5 bullet points summarizing the condition of the dataset.

Focus on what you found from profiling. Do **not** clean the data yet.

Examples:
- The dataset has ___ rows and ___ columns.
- The ___ column has missing values.
- The ___ column may have an incorrect data type.
- The ___ column has possible duplicate IDs.
- The ___ column has a range from ___ to ___.


**Your observations:**

1. The dataset has 13 rows and 8 columns.
2. The hire_date, salary, age, and full_time columns each contain a missing value.
3. The hire_date may have an incorrect data type as it is stored as str instead of a data date type.
4. The employee_id column has possible duplicate IDs.
5. The age column has a range from 28 to 999.


## Submission Checklist
Before submitting, make sure:

- [yes] All code cells run without errors.
- [yes] You completed all tasks.
- [yes] You wrote short responses in the markdown cells.
- [yes] You saved the notebook.
- [yes] You committed and pushed your work to GitHub.

Use the terminal:

```bash
git add .
git commit -m "Completed Week 2 data profile checkpoint"
git push
```
